src/Symfony/Component/Security/Http/Authenticator/FormLoginAuthenticator.php

```diff
@@ -146,4 +148,24 @@ private function getCredentials(Request $request): array
 
         return $credentials;
     }
+
+    public function setHttpKernel(HttpKernelInterface $httpKernel): void
+    {
+        $this->httpKernel = $httpKernel;
+    }
+
+    public function start(Request $request, AuthenticationException $authException = null): Response
+    {
+        if (!$this->options['use_forward']) {
+            return parent::start($request, $authException);
+        }
+
```

Comment:
We probably need a sanity check here - check of `$this->httpKernel` is set and throw an exception if it is not.

Or... another option is to not support `use_forward` in the new system to avoid this messiness...

Reply:
I would vote to leave it like this: when using the Symfony framework, there is no way to end up here without `$this->httpKernel` being set.

I plan to deprecate this feature in Symfony 5.3, so we don't need to be too strictly nice here imho.